scripts/ollama-setup.sh

```
@@ -0,0 +1,377 @@
+        else
+            print_warning "Failed to install $model, continuing..."
+        fi
+    done
+}
+
+# Function to check system requirements
+check_system_requirements() {
+    print_status "Checking system requirements..."
+    
+    # Check available memory
+    local total_mem=$(free -m 2>/dev/null | awk 'NR==2{printf "%.0f", $2/1024}' || echo "0")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)